src/batch.rs

```diff
@@ -120,59 +186,107 @@ impl<T: SigType> Verifier<T> {
+                    // let tup: BatchTuple<SpendAuth> = (vk_bytes, sig, c);
 
-            let s = {
-                // XXX-jubjub: should not use CtOption here
-                let maybe_scalar = Scalar::from_bytes(&sig.s_bytes);
-                if maybe_scalar.is_some().into() {
-                    maybe_scalar.unwrap()
-                } else {
-                    return Err(Error::InvalidSignature);
+                    // let (P_coeff, R, R_coeff, VK, VK_coeff) =
+                    //     &self.compute_multiscalar_mul_inputs(rng, tup)?;
+
+                    let z = Scalar::from_raw(gen_128_bits(&mut rng));
+
+                    let s = {
+                        // XXX-jubjub: should not use CtOption here
+                        let maybe_scalar = Scalar::from_bytes(&sig.s_bytes);
+                        if maybe_scalar.is_some().into() {
+                            maybe_scalar.unwrap()
+                        } else {
+                            return Err(Error::InvalidSignature);
+                        }
+                    };
+
+                    let R = {
+                        // XXX-jubjub: should not use CtOption here
+                        // XXX-jubjub: inconsistent ownership in from_bytes
+                        let maybe_point = AffinePoint::from_bytes(sig.r_bytes);
+                        if maybe_point.is_some().into() {
+                            jubjub::ExtendedPoint::from(maybe_point.unwrap())
+                        } else {
+                            return Err(Error::InvalidSignature);
+                        }
+                    };
+                    let P_coeff = z * s;
+                    let R_coeff = z;
+                    let VK = VerificationKey::<SpendAuth>::try_from(vk_bytes.bytes)
+                        .unwrap()
+                        .point;
+                    let VK_coeff = Scalar::zero() + (z * c);
+
+                    P_spendauth_coeff -= P_coeff;
+                    Rs.push(R);
+                    R_coeffs.push(R_coeff);
+                    VKs.push(VK);
+                    VK_coeffs.push(VK_coeff);
                 }
-            };
+                Inner::Binding { vk_bytes, sig, c } => {
+                    // let tup: BatchTuple<Binding> = (vk_bytes, sig, c);
 
-            let z = Scalar::from_raw(gen_128_bits(&mut rng));
+                    // let (P_coeff, R, R_coeff, VK, VK_coeff) =
+                    //     &self.compute_multiscalar_mul_inputs(rng, tup)?;
 
-            P_coeff -= z * s;
+                    let z = Scalar::from_raw(gen_128_bits(&mut rng));
 
-            Rs.push(R);
-            R_coeffs.push(z);
+                    let s = {
+                        // XXX-jubjub: should not use CtOption here
+                        let maybe_scalar = Scalar::from_bytes(&sig.s_bytes);
+                        if maybe_scalar.is_some().into() {
+                            maybe_scalar.unwrap()
```

Comment:
Note: here and below rather than `unwrap`, it should propagate errors.

Reply:
So not catching it and then returning `Err(Error::InvalidSignature)` ?

Reply:
No, I was wrong and missed the `is_some()` check.

Reply:
this is just a CtOption / jubjub api issue